sonic_platform_base/sfp_base.py

```diff
@@ -0,0 +1,204 @@
+#
+# sfp_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a SFP module in SONiC
+#
+
+import sys
+from . import device_base
+
+
+class SfpBase(device_base.DeviceBase):
+    """
+    Abstract base class for interfacing with a SFP module
+    """
+
+    def get_transceiver_info(self):
+        """
+        Retrieves transceiver info of this SFP
+
+        Returns:
+            A dict which contains following keys/values :
+        ========================================================================
+        keys                       |Value Format   |Information	
+        ---------------------------|---------------|----------------------------
+        type                       |1*255VCHAR     |type of SFP
+        hardwarerev                |1*255VCHAR     |hardware version of SFP
+        serialnum                  |1*255VCHAR     |serial number of the SFP
+        manufacturename            |1*255VCHAR     |SFP venndor name
+        modelname                  |1*255VCHAR     |SFP model name
+        Connector                  |1*255VCHAR     |connector information
+        encoding                   |1*255VCHAR     |encoding information
+        ext_identifier             |1*255VCHAR     |extend identifier
+        ext_rateselect_compliance  |1*255VCHAR     |extended rateSelect compliance
+        cable_length               |INT            |cable length in m
```

Comment:
a cable length may not be able to reflect all the info in the eeprom, in sff8436 or sff8636 we can see that several field to reflect different cable type, like Length(SMF) for length supported for SMF fiber in km, Length(OM3 50um) for length supported for EBW 50/125 um fiber, units of 2m, Length(copper), etc.

Reply:
I don't see SONiC being used in long haul, most of them are in data center.  Use km as unit might be an overkill.   For now, I would recommend that we keep it at m.   Let me know your thought.